lib/permissions.js

```diff
@@ -128,61 +128,33 @@ function PolicyStatementParser(action, resource, statements, callback) {
      *
      * @callback cb - function of the form function(error, allowed)
      */
-
     parse: function (cb) {
-      var cachedPermissions = app.usermanager.getSessionVariable('cachedPermissions') || {};
-      if ('undefined' !== typeof cachedPermissions[resource]) {
-        return cb(null, cachedPermissions[resource]);
-      }
+      let allowed = false;
+      const resourceToken = this.tokens(resource);
 
-      var index, el;
-      var allowed = false;
-
-      var resourceTok = this.tokens(resource);
-      if (!resourceTok) {
-        callback(new Error('given resource is not a valid format'));
+      if (!resourceToken) {
+        return callback(new Error('given resource is not a valid format'));
       }
-
-      for (index = 0; index < statements.length; ++index) {
-        el = statements[index];
-        if (!el) { // save us!
-          continue;
-        }
-
-        // check if action matches
-        if (-1 === el.action.indexOf(action)) {
-          // no point in further parsing of the statement
+      for(let i = 0, statement = statements[i]; i < statements.length; statement = statements[++i]) {
```

Comment:
As discussed, might be more readable keeping the `statement` assignment in the body.